client/packages/programs/src/JsonForms/components/Search/SearchWithUserSource.tsx

```diff
@@ -80,16 +84,19 @@ export const SearchWithUserSource = (props: ControlProps) => {
   const handleDataUpdate = (selectedResult: Record<string, any> | null) => {
     if (selectedResult === null) return;
     if (!saveFields) handleChange(path, selectedResult);
-
-    const newObj: Record<string, any> = {};
-    saveFields?.forEach(
-      field => (newObj[field] = selectedResult[field] ?? null)
-    );
-    handleChange(path, newObj);
+    else {
+      const newObj: Record<string, any> = {};
+      saveFields?.forEach(
+        field => (newObj[field] = selectedResult[field] ?? null)
+      );
+      handleChange(path, newObj);
+    }
     setEditMode(false);
   };
 
-  if (!options) return null;
+  const error = zErrors ?? props.errors ?? queryError ?? null;
+
+  if (!visible) return null;
```

Comment:
I noticed quite a few of the existing components aren't handling `visible` (or `enabled`), so have made an issue to check and fix these: #1228